src/test/java/org/llorllale/cactoos/matchers/Assertion2Test.java

```diff
@@ -0,0 +1,158 @@
+import org.cactoos.text.TextOf;
+import org.hamcrest.core.IsEqual;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Tests for {@link Assertion}.
+ *
+ * @since 1.0.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class Assertion2Test {
+    /**
+     * A rule for handling an exception.
+     */
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    /**
+     * Assertion can be affirmed if the operation being tested matches.
+     */
+    @Test
+    public void affirmIfResultMatches() {
+        final String expected = "abc123";
+        new Assertion2<>(
+            "must affirm the assertion if the test's result is as expected",
+            new TextOf(expected),
+            new TextIs(expected)
+        ).affirm();
+    }
+
+    /**
+     * Assertion must be refuted if the operation being tested does not
+     * match.
+     */
+    @Test(expected = AssertionError.class)
+    public void refuteIfResultDoesNotMatch() {
+        new Assertion2<>(
+            "must refute the assertion if the test's result is not as expected",
+            new TextOf("test"),
+            new TextIs("no match")
+        ).affirm();
+    }
+
+    /**
+     * Assertion cannot be affirmed if the operation being tested throws an
+     * unexpected error.
+     */
+    @Test(expected = IllegalStateException.class)
+    public void refuteIfErrorDoesNotMatch() {
+        new Assertion2<>(
+            "must fail if the test throws an unexpected error",
+            () -> {
+                throw new IllegalStateException();
+            },
+            new TextIs("no match")
+        ).affirm();
+    }
+
+    /**
+     * Assertion can be affirmed if the operation being tested throws an
+     * expected error.
+     */
+    @Test
+    public void affirmIfErrorMatches() {
+        new Assertion2<>(
+            "must affirm the assertion if the test throws the expected error",
+            () -> {
+                throw new IllegalStateException("this is a test");
+            },
+            new Throws<>("this is a test", IllegalStateException.class)
+        ).affirm();
+    }
+
+    /**
+     * The scalar within Assertion executed is only once.
+     */
+    @Test
+    public void scalarIsExecutingOnce() {
+        final AtomicInteger quantity = new AtomicInteger(0);
+        new Assertion2<>(
+            "scalar throws the exception",
+            () -> {
+                quantity.incrementAndGet();
+                throw new IllegalStateException("this is a test");
+            },
+            new Throws<>("this is a test", IllegalStateException.class)
+        ).affirm();
+        new Assertion2<>(
+            "scalar is executing once",
+            quantity.get(),
+            new IsEqual<>(1)
+        ).affirm();
+    }
+
+    @Test
+    public void testIsExecutedOnlyOnce() {
```

Comment:
@llorllale I think this test does not make any sense any more since we don't pass a `Scalar`

Reply:
@victornoel also, `scalarIsExecutingOnce` and `testWithThrowsIsExecutedOnlyOnce` are duplicates for some reason... will remove one